perf/perf_rawevents.py

```diff
@@ -86,17 +93,24 @@ def error_check(self):
                 self.log.info("Failed command: %s", self.fail_cmd[cmd])
             self.fail("perf_raw_events: refer log file for failed events")
 
-    def test_raw_code(self):
-        file_name = 'raw_codes_p' + self.cpu_family
-        perf_flags = "perf stat -e r"
-        self.run_event(file_name, perf_flags)
-        self.error_check()
-
-    def test_name_event(self):
-        file_name = 'name_events_p' + self.cpu_family
-        perf_flags = "perf stat -e "
-        self.run_event(file_name, perf_flags)
-        self.error_check()
+    def test(self):
+        if '004b' in self.rev:
+            self.run_event('raw_codes_p8', 'raw')
+            self.error_check()
+            self.run_event('name_events_p8', 'name')
+            self.error_check()
+        if '004e' in self.rev:
+            self.run_event('raw_codes_p9', 'raw')
+            self.error_check()
+            self.run_event('name_events_p9', 'name')
+            self.error_check()
+        if '0080' in self.rev:
+            self.run_event('raw_codes_p10', 'raw')
+            self.error_check()
+            self.run_event('name_events_p10', 'name')
+            self.error_check()
```

Comment:
I think we need to simplify this or else we will end up adding more if loops on getting new chips/servers :-)
I suggest use for loop and dictionary.
cpu_dic = {'004b':'p8', '004e':'p9', '0080':'p10'}
then run the for_loop on dic. 
in future if we get new chips adding an entry to dic will be enough.
hope this will help you :-)